e2e/tests/dump-restore-test.rs

```diff
@@ -1,38 +1,65 @@
-use similar_asserts::assert_str_eq;
 use std::env;
 use std::fs;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::process::Command;
+use std::vec::Vec;
 use testcontainers::clients::Cli;
 use testcontainers::images::generic::{GenericImage, WaitFor};
 use testcontainers::{clients, Container};
 use testcontainers::{images, Docker};
 
-fn run_postgres<'a>(docker: &'a Cli, db: &str, username: &str) -> Container<'a, Cli, GenericImage> {
-    //let docker_image = env::var("TS_DOCKER_IMAGE").unwrap_or_else(|_| {
-    //    String::from("ghcr.io/timescale/dev_promscale_extension:develop-ts2-pg14")
-    //});
-    // todo: hardcoded for now. fix when the extension is installed in public schema instead of _prom_ext
-    let docker_image = "ghcr.io/timescale/dev_promscale_extension:jg-sec-audit-prom-ext-ts2-pg14";
+/// A docker container running Postgres
+type PostgresContainer<'d> = Container<'d, Cli, GenericImage>;
 
-    let src = concat!(env!("CARGO_MANIFEST_DIR"), "/scripts");
-    let dest = "/scripts";
+/// Returns the name of the docker image to use for Postgres containers.
+/// If the `TS_DOCKER_IMAGE` environment variable is set, it will return that value.
+/// Otherwise, it returns a default image.
+fn postgres_image() -> String {
+    env::var("TS_DOCKER_IMAGE").unwrap_or_else(|_| {
+        String::from("ghcr.io/timescale/dev_promscale_extension:develop-ts2-pg14")
+    })
+}
 
-    let generic_postgres = images::generic::GenericImage::new(docker_image)
+/// Creates and runs a docker container running Postgres
+fn run_postgres<'a>(
+    docker: &'a Cli,
+    docker_image: &str,
+    db: &str,
+    username: &str,
+    volumes: Option<&Vec<(&str, &str)>>,
+) -> PostgresContainer<'a> {
+    let mut generic_postgres = images::generic::GenericImage::new(docker_image)
         .with_wait_for(WaitFor::message_on_stderr(
             "database system is ready to accept connections",
         ))
         .with_env_var("POSTGRES_DB", db)
         .with_env_var("POSTGRES_USER", username)
         .with_env_var("POSTGRES_HOST_AUTH_METHOD", "trust")
-        .with_env_var("POSTGRES_PASSWORD", "password")
-        .with_volume(src, dest);
+        .with_env_var("POSTGRES_PASSWORD", "password");
+
+    if let Some(volumes) = volumes {
+        for volume in volumes.iter() {
+            generic_postgres = generic_postgres.with_volume(volume.0, volume.1);
+        }
+    }
 
     docker.run(generic_postgres)
 }
 
-fn psql_script(container: &Container<Cli, GenericImage>, db: &str, username: &str, path: &str) {
-    println!("executing psql script {}...", path);
+/// Runs a SQL script file in the docker container
+///
+/// The script file is copied into the container, and executed with
```

Comment:
This says that the script file is copied into the container, but I don't see that actually happening in the implementation of `psql_file`.

Reply:
fixed in 834df2de0c02767260f924e0766a29384e996f39